backend/routes/portfolios.js

```diff
@@ -138,18 +138,18 @@ router.put("/addMany/:companyId", async (req, res, next) => {
   const companyId = parseInt(req.params.companyId);
   const possibleIds = req.body.ids;
 
-  const portfolio = await prisma.portfolio.findUnique({
-    where: {
-      id: portfolioId,
-    },
-  });
-
-  if (possibleIds == null) {
-    next(BadParams("no ids specified, or other bad param issue"));
+  if (companyId == null) {
+    next(BadParams("companyids specified, or other bad param issue"));
   }
+
   const ids = possibleIds.map((val) => parseInt(val));
 
-  for (let portfolioIds of ids) {
+  for (let portfolioId of ids) {
```

Comment:
instead of query portfolio one by one, can you use `findMany` to query all portfolios and loop prisma object instead? e.g.
```
const portfolios = await prisma.portfolio.findMany({where: {
        id: { in: ids },
      },
for(let portfolio of portfolios) {
... call update here
}
```